client/js/analysis.js

```diff
@@ -1,25 +1,113 @@
+const COMMENT_APPEARANCE_TIME = 1500;
+const COMMENT_TO_STOP_AT = 5;
+
 window.onload = function() {
-  for (let i = 1; i < 6; i++) {
+  // Listen for submission click
+  const formSubmit = document.getElementById('sendbutton');
+  formSubmit.addEventListener('click', fetchResponse, false);
+};
+
+/**
+ * Fetches the data from the backend and populates the screen
+ *
+ */
+async function fetchResponse() {
+  const videoQuery = document.getElementById('searchbar').value;
+  const response = await fetch(`/api/analysis?name=${videoQuery}`);
+  const responseJson = await response.json();
+  populationHandler(responseJson);
+}
+
+/**
+ * Handles populating all elements on the screen
+ *
+ * @param {object} videoAnalysis the object returned from the HTTP request
+ * which contains all of the data.
+ */
+function populationHandler(videoAnalysis) {
+  const charts = document.getElementById('charts');
+  const list = document.getElementById('list');
+
+  removeAllChildNodes(charts);
+  removeAllChildNodes(list);
+  populateDonutCharts(videoAnalysis.perspectiveMap);
+  // The +1 allows for determineOverall() to run correctly.
```

Comment:
good comment

Reply:
Minimally helpful comment actually. Describe what the +1 does and how it interacts with `determineOverall`. Especially since `time` isn't passed to `determineOverall` anywhere.

It seems to be just used in the setTimeout below. I suggest restructuring this as:
```
const totalComments = renderComments(...);
const commentsRenderTime = totalComments * COMMENT_APPEARANCE_TIME;
...
setTimeout(() => {
  ...
}, commentsRenderTime + FEEDBACK_APPEARANCE_TIME);
```